File: app/src/main/java/com/raredev/vcspace/ui/editor/symbol/SymbolInputView.java
package com.raredev.vcspace.ui.editor.symbol;

import android.content.Context;
import android.util.AttributeSet;
import androidx.annotation.NonNull;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;
import com.raredev.vcspace.ui.editor.Symbol;
import com.raredev.vcspace.ui.editor.symbol.adapter.SymbolInputAdapter;
import com.raredev.vcspace.util.PreferencesUtils;
import io.github.rosemoe.sora.widget.CodeEditor;
import java.util.ArrayList;
import java.util.List;

public class SymbolInputView extends RecyclerView {
  private SymbolInputAdapter adapter;

  public SymbolInputView(Context context) {
    this(context, null);
  }

  public SymbolInputView(Context context, AttributeSet attrs) {
    this(context, attrs, 0);
  }

  public SymbolInputView(Context context, AttributeSet attrs, int defStyle) {
    super(context, attrs, defStyle);
    setLayoutManager(new LinearLayoutManager(getContext(), LinearLayoutManager.HORIZONTAL, false));
    adapter = new SymbolInputAdapter();
    setAdapter(adapter);

    setSymbols();
  }

  public void bindEditor(@NonNull CodeEditor editor) {
    adapter.bindEditor(editor);
  }

  public void setSymbols() {
    List<Symbol> symbols = new ArrayList<>();
    symbols.add(new Symbol("→", PreferencesUtils.useUseSpaces() ? "    " : "\t"));
    symbols.add(new Symbol("\""));
    symbols.add(new Symbol(";"));
    symbols.add(new Symbol("(", "()"));
    symbols.add(new Symbol(")"));
    symbols.add(new Symbol("{", "{}"));
    symbols.add(new Symbol("}"));
    symbols.add(new Symbol("[", "[]"));
    symbols.add(new Symbol("]"));
    symbols.add(new Symbol("<", "<>"));
    symbols.add(new Symbol(">"));

    adapter.setSymbols(symbols);
  }
}

File: app/src/main/java/com/teixeira/vcspace/ui/SymbolInputView.kt
package com.teixeira.vcspace.ui

import android.content.Context
import android.util.AttributeSet
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.teixeira.vcspace.adapters.SymbolInputAdapter
import com.teixeira.vcspace.editor.VCSpaceEditor

class SymbolInputView : RecyclerView {

  constructor(context: Context) : super(context)

  constructor(context: Context, attrs: AttributeSet?) : super(context, attrs)

  constructor(
    context: Context,
    attrs: AttributeSet?,
    defStyleAttr: Int
  ) : super(context, attrs, defStyleAttr)

  private val adapter = SymbolInputAdapter()

  init {
    layoutManager = LinearLayoutManager(context, LinearLayoutManager.HORIZONTAL, false)
    setAdapter(adapter)
  }

  fun bindEditor(editor: VCSpaceEditor?) {
    adapter.bindEditor(editor)
  }
}
